File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/inspection-order/checklist/property/high-value-bath.ts
import { InspectionOrderProperty } from "../property";
import { InspectionOrderPropertyHighValueBathBathroomCounters } from "./high-value-bath/InspectionOrderPropertyHighValueBathBathroomCounters";
import { InspectionOrderPropertyHighValueBathBathroomFixtures } from "./high-value-bath/InspectionOrderPropertyHighValueBathBathroomFixtures";
import { InspectionOrderPropertyHighValueBathBathroomFloors } from "./high-value-bath/InspectionOrderPropertyHighValueBathBathroomFloors";
import { InspectionOrderPropertyHighValueBathBathroomVanities } from "./high-value-bath/InspectionOrderPropertyHighValueBathBathroomVanities";
import { InspectionOrderPropertyHighValueBathTubsAndShowers } from "./high-value-bath/InspectionOrderPropertyHighValueBathTubsAndShowers";

export class InspectionOrderPropertyHighValueBath {
    /*[Parent]*/
    inspectionOrderProperty?: InspectionOrderProperty;
    /*[Key]*/
    id?: string;
    numberofFullBath?: boolean;
    numberofHalfBath?: boolean;
    bathroomFloors?: InspectionOrderPropertyHighValueBathBathroomFloors[];
    bathroomVanities?: InspectionOrderPropertyHighValueBathBathroomVanities[];
    bathroomCounters?: InspectionOrderPropertyHighValueBathBathroomCounters[];
    bathroomFixtures?: InspectionOrderPropertyHighValueBathBathroomFixtures[];
    tubsAndShowers?: InspectionOrderPropertyHighValueBathTubsAndShowers[];
    public constructor(init?:Partial<InspectionOrderPropertyHighValueBath>) {
        Object.assign(this, init);
    }
  }

File: Rivington.IG.Web/Rivington.IG.Domain.Models/OrderManagement/InspectionOrderProperty/HighValueBath.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace Rivington.IG.Domain.Models.OrderManagement
{
    public class InspectionOrderPropertyHighValueBath
    {
        [Parent]
        public InspectionOrderProperty InspectionOrderProperty { get; set; }
        [Key]
        public Guid Id { get; set; }

        public int? NumberofFullBath { get; set; }
        public int? NumberofHalfBath { get; set; }
        public ICollection<InspectionOrderPropertyHighValueBathBathroomFloors> BathroomFloors { get; set; }
        public ICollection<InspectionOrderPropertyHighValueBathBathroomVanities> BathroomVanities { get; set; }
        public ICollection<InspectionOrderPropertyHighValueBathBathroomCounters> BathroomCounters { get; set; }
        public ICollection<InspectionOrderPropertyHighValueBathBathroomFixtures> BathroomFixtures { get; set; }
        public ICollection<InspectionOrderPropertyHighValueBathTubsAndShowers> TubsAndShowers { get; set; }

        public static void BuildModel(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<InspectionOrderProperty>()
                .HasOne(a => a.HighValueBath).WithOne(b => b.InspectionOrderProperty)
                .HasForeignKey<InspectionOrderPropertyHighValueBath>()
                .HasConstraintName($"FK_{nameof(InspectionOrderProperty)}_{nameof(InspectionOrderPropertyHighValueBath)}")
                .OnDelete(DeleteBehavior.Cascade);

            InspectionOrderPropertyHighValueBathBathroomFloors.BuildModel(modelBuilder);

            InspectionOrderPropertyHighValueBathBathroomVanities.BuildModel(modelBuilder);

            InspectionOrderPropertyHighValueBathBathroomCounters.BuildModel(modelBuilder);

            InspectionOrderPropertyHighValueBathBathroomFixtures.BuildModel(modelBuilder);

            InspectionOrderPropertyHighValueBathTubsAndShowers.BuildModel(modelBuilder);
        }
    }
}